analyzer.go

```diff
@@ -1,50 +1,54 @@
 package lifecycle
 
 import (
-	"fmt"
-	"path/filepath"
-
 	"github.com/buildpacks/imgutil"
 	"github.com/pkg/errors"
 
+	"github.com/buildpacks/lifecycle/api"
 	"github.com/buildpacks/lifecycle/buildpack"
-	"github.com/buildpacks/lifecycle/launch"
+	"github.com/buildpacks/lifecycle/cmd"
 	"github.com/buildpacks/lifecycle/platform"
 )
 
 type Analyzer struct {
-	Buildpacks []buildpack.GroupBuildpack
-	LayersDir  string
-	Logger     Logger
-	SkipLayers bool
+	Image         imgutil.Image
+	LayersDir     string
+	Logger        Logger
+	LayerAnalyzer LayerAnalyzer
+	Platform      cmd.Platform
+	Buildpacks    []buildpack.GroupBuildpack
+	SkipLayers    bool
+
+	// Platform API < 0.7
+	Cache Cache
 }
 
-// Analyze restores metadata for launch and cache layers into the layers directory.
-// If a usable cache is not provided, Analyze will not restore any cache=true layer metadata.
-func (a *Analyzer) Analyze(image imgutil.Image, cache Cache) (platform.AnalyzedMetadata, error) {
-	imageID, err := a.getImageIdentifier(image)
-	if err != nil {
-		return platform.AnalyzedMetadata{}, errors.Wrap(err, "retrieving image identifier")
-	}
+// Analyze fetches the layers metadata from the previous image and writes analyzed.toml
```

Comment:
```suggestion
// Analyze fetches the layers metadata from the previous image and writes analyzed.toml.
```

Reply:
I wonder if it's worth describing the per-API behavior in this comment.

Reply:
I don't feel strongly enough that this documentation would benefit from that, but we should chat if you feel strongly.

Reply:
Ehh... not strongly. I was thinking something short and sweet like "For earlier platform APIs, analyzer also writes layer metadata files in the layers directory."